src/observer.jl

```diff
@@ -1,72 +1,117 @@
 """
-TomographyObserver is an implementation of an
+Observer is an implementation of an
 observer object (<:AbstractObserver) which
-implements custom measurements.
+implements custom measurements to perform
+at each layer of the circuit evolution.
 """
-struct TomographyObserver <: AbstractObserver
-  fidelity::Vector{Float64}
-  train_loss::Vector{Float64}
-  test_loss::Vector{Float64}
-  trace_preserving_distance::Vector{Float64}
-  frobenius_distance::Vector{Float64}
-  fidelity_bound::Vector{Float64}
-  function TomographyObserver()
-    return new([],[],[],[],[],[])
-  end
+struct Observer <: AbstractObserver
+  measurements::Dict{String, Pair{<:Union{Nothing,Function, String, Tuple, Pair{<:Function, <:Any}},<:Any}}
```

Comment:
What measurement would have a type `Pair{<:Function, <:Any}`? Could it just be covered by a `Function`?

Reply:
I think I see that it is a way to have a custom `measure` function, something like `push!(obs, mymeasure => ("X", 2))`. It still seems like this could be handled by using global variable, like:
```julia
function mymeasure(M::MPS, observable::Tuple)
  [measure `observable`...]
end
mymeasure_X2(M::MPS) = mymeasure(M, ("X", 2))
# Instead of:
# push!(obs, mymeasure => ("X", 2))
push!(obs, mymeasure_X2)
```

Reply:
I will explain the details in our call